### DISCUSSION TITLE
[Question] Enumerating DPIDs from the Tyua MCU

### DISCUSSION THREAD

**Discussion by jwaz73** [jazzmonger/wood-pellet-stove-with-TYWE1S-Tuya-chip#18] [General]
First of all, thank you for taking the time to document your efforts on this issue!  This is very beneficial.

A question for you.  How did you enumerate the DPIDs and Power Mode / ECO Mode maps you have in the ESPHome yaml file?  

The only way I have been able to see DPID data is through the LocalTuya integration in HA.  I have looked and looked for the integer IDs in the Tuya IoT console but can only find the text representation.  I have not found anything that shows me the 'actual' value of the power or eco mode settings.  P1 = 0 for example...

I'm working with a Master Forge H140XL and the controller looks very similar to the photos you posted. Only difference seems to be the enclosure (mine is built-in to the side of the stove).

I am seeing an additional DPID from what you have documented.

105 (Value: F)

I'm thinking this is an indicator of the temperature measurement unit (Fahrenheit)...

Thanks again for sharing your hard-learned knowledge on these pellet stove controllers!

-James

**Comment by jazzmonger** [comment]
Dude.., I've been working on this for over a year! Glad you can use my learnings. It all came together once I flashed the esp a week ago.  then I kicked it into high gear.  I just posted all this a couple days ago bc I was sure somebody somewhere out in the universe would need it. And as time goes by, I was sure I'd forget something!

Did yours suffer the same Tuya temp reporting problem before you flashed it? Or have you flashed it yet?

Datapoints... I used this method. It's for zigbee but it works for ours. 

https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html#_9-find-your-data-point

And I also have a dp5. It doesn't change or do anything useful that I can tell. 

Send me a pic of your controller and I'll include your model in the readme.  Jazzmonger@gmail.com

Here's my latest card I created that gives me everything in one shot:


![E000CDA3-F8B0-4B1C-9062-77204C9AE41B](https://user-images.githubusercontent.com/52110065/202742341-fc60d783-9eb3-463e-9292-e563b9b13e66.jpeg)

**Comment by jazzmonger** [comment]
Your stove is the exact same one I have in my shop.  It has the same controller. Just a rebrand of a Chinese stove.  In fact, Masterforge has the same stove I have in my house as well.
Check it out 
https://www.cleveland-ironworks.com/products/pellet-stoves.html

I updated the readme with this info.

I have terrible overheating problems with that stove on P1 and it just shuts down. I fixed it by putting a fan behind it.  I'll flash it's controller next.

**Comment by jwaz73** [comment]
I did have the 'unexpected shutdown' issue a couple of times.  Both times my display has shown me an E6 error code indicating the temp sensor below the pellet hopper is reading too high.  The stove was on P1 mode for both of these.  We do suffer regular power blinks here and each time the stove restarts from a power failure it defaults to the P1 mode.  Now that I have a path to better integration with HA I will address the power setting with automation.  I haven't flashed it yet.  It's the only heat source for my workshop so I'm waiting on a second display panel to arrive from the manufacturer before modifying.

**Comment by jazzmonger** [comment]
If you adjust the combustion blower to -20 on P1 thru the menu setting, and that can also help eliminate that problem.  I run your stove also in my shop and set it on ECO2.  It heats on PI then adjusts down to P4 and back to P1 when it needs to.

the worst part about the errors is they don't clear.  At first I thought you had to power cycle the whole stove to clear them,  but I recently figured out (by reading the manual for the Nemaxx stove - its VERY detailed)  that if you press and hold thumbs up, it will also clear the error.  But unfortunately, there is no way I've found (yet) to clear an error condition programatically.  it might be possible to take an unused GPIO pin off of the TYWE1S and wire it into the thumbs up button, but for now I just use a wifi outlet and power cycle that.

The good news is the TYWE1S module can either be installed, or not.  Its just a daughterboard to the main MCU.  It just relays msgs to and from the main MCU.  In fact, my vendor sent me 2 displays w/out the wifi modules by mistake.

One of my "down time" projects I'm contemplating is wiring into the display to see what it's doing at all times.  I just ordered a 16ch logic probe from China for like $60.  Tons of sw for them too.
